src/main/java/com/danielagapov/spawn/Services/User/UserService.java

```diff
@@ -396,24 +396,38 @@ public List<RecommendedFriendUserDTO> getRecommendedFriendsForUserId(UUID userId
             // Fetch the requesting user's friends
             List<UUID> requestingUserFriendIds = getFriendUserIdsByUserId(userId);
 
-            // Create a set of the requesting user's friends for quick lookup
-            Set<UUID> requestingUserFriendSet = new HashSet<>(requestingUserFriendIds);
+            // Fetch users who have already received a friend request from the user
+            List<UUID> sentFriendRequestIds = friendRequestService.getSentFriendRequestsByUserId(userId)
+                    .stream()
+                    .map(FriendRequestDTO::receiverUserId)
+                    .collect(Collectors.toList());
+
+            // Fetch users who have sent a friend request to the user (pending requests)
+            List<UUID> receivedFriendRequestIds = friendRequestService.getIncomingFriendRequestsByUserId(userId)
```

Comment:
```suggestion
            List<UUID> receivedFriendRequestSenderUserIds = friendRequestService.getIncomingFriendRequestsByUserId(userId)
```
Same as above, for clarity